### PR TITLE
fix: make crosschain-call with invalid withdraw revert

### PR DESCRIPTION
**Pull request by lumtis**
# Description

Solves https://github.com/zeta-chain/node/issues/3308

If a crosschain call initiate an invalid withdraw (example dust amount for Bitcoin withdraw) makes the CCXT reverts

**Solution**

In `HandleEVMCall` where crosschain calls are handled, when `ProcessLogs` logs fails, we consider that the crosschain call reverts because the withdraw initiated in this call can't be processed, we therefore returns `contractReverted == true` so it enters the revert workflow.

Since in this case the deposit contract call was successful and there is an effective state transition in the EVM, we now use `tmpCtx`, we only commit it if the contract has a regular revert or the call is sucessful and `ProcessLogs` doesn't fail.

The E2E test checks that the BTC balance remains 0 after the calls, this means the EVM state transition has not been processed.

## Release Notes

- **New Features**
	- Introduced a new telemetry endpoint `/systemtime` in the application.
	- Added a crosschain-call feature to handle invalid withdraw reverts.
	- New simulation tests for custom zetachain modules and performance tests for Solana E2E.

- **Bug Fixes**
	- Fixed issues with unsupported Solana transaction versions and inbound vote message validations.
	- Updated handling of pending nonces during transaction abortion.

- **Documentation**
	- Updated changelog to reflect new features, tests, and fixes.

- **Tests**
	- Added a new test for Bitcoin deposit and withdrawal operations involving dust amounts.

### CODE REVIEW

File: e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// It tests the edge case where during a cross-chain call, a invaild withdraw is initiated (processLogs fails)
+func TestBitcoinDepositAndWithdrawWithDust(r *runner.E2ERunner, args []string) {
+	// Given "Live" BTC network
+	stop := r.MineBlocksIfLocalBitcoin()
+	defer stop()
+
+	require.Len(r, args, 0)
+
+	// ARRANGE
+
```
Comment (ws4charlie) [inline e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:25]:
```suggestion
	// ARRANGE
```

Comment (lumtis) [reply e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:25]: What's that?

Comment (ws4charlie) [reply e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:25]: extra new line

File: e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	// ASSERT
+	// Now we want to make sure the cctx is reverted with expected error message
+
+	// cctx status would be pending revert if using v21 or before
+	cctx := utils.WaitCctxRevertedByInboundHash(r.Ctx, r, txHash.String(), r.CctxClient)
+	require.Equal(r, crosschaintypes.CctxStatus_Reverted, cctx.CctxStatus.Status)
+	require.Contains(r, cctx.CctxStatus.ErrorMessage, crosschaintypes.ErrCannotProcessWithdrawal.Error())
+
+	// check the contract has no BTC balance, this would mean the contract call state transition is not reverted
```
Comment (ws4charlie) [inline e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:53]: the expected behavior is `not reverted` or `reverted`?

File: x/crosschain/keeper/evm_deposit.go [hunk 3]
```diff
@@ -145,7 +154,11 @@ func (k Keeper) HandleEVMDeposit(ctx sdk.Context, cctx *types.CrossChainTx) (boo
 				)
 			}
 		}
+
```
Comment (kingpinXD) [inline x/crosschain/keeper/evm_deposit.go:157]: Consider putting lines 105 - 157 in a block , `{ }` 
For readability to show, where the context is branched and committed back

**Review by kingpinXD — APPROVED** [review]
Great catch , PR looks good to me

File: e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	require.Equal(r, receipt.Status, uint64(1))
+
+	// Given a list of UTXOs
+	utxos, err := r.ListDeployerUTXOs()
+	require.NoError(r, err)
+	require.NotEmpty(r, utxos)
+
+	// ACT
+	// Deposit 0.01 BTC to withdrawer, this is an arbitrary amount, must be greater than dust amount
+	txHash, err := r.SendToTSSFromDeployerWithMemo(0.01, utxos, withdrawerAddr.Bytes())
```
Comment (ws4charlie) [inline e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:41]: This looks to me a pure `deposit` of 0.01 BTC. How would the `onCrossChainCall` on withdrawer contract be called in this scenario? a bit confused

File: e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go (hunk 1 above)
Comment (lumtis) [reply e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:25]: There is no change in your suggestion

File: e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go (hunk 2 above)
Comment (lumtis) [reply e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:53]: Need to make the comment more clear

File: e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go (hunk 4 above)
Comment (lumtis) [reply e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:41]: Mmh weird, the logs shows smart contract call happens

Comment (lumtis) [reply e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:41]: Ok, just checked, independently of the message, we do a contract call if dest is a contract in the V1 workflow

File: x/crosschain/keeper/evm_deposit.go (hunk 3 above)
Comment (lumtis) [reply x/crosschain/keeper/evm_deposit.go:157]: I don't think you should use block if it doesn't have reason in the workflow

File: e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go (hunk 4 above)
Comment (ws4charlie) [reply e2e/e2etests/test_bitcoin_deposit_and_withdraw_with_dust.go:41]: okay. That makes sense then.

**Review by ws4charlie — APPROVED** [review]
lgtm

**Comment by lumtis** [comment]
Received a security review
